src/Api/IngestionBase.php

```diff
@@ -103,16 +103,16 @@ public function getUri(): string
     /**
      * @inheritDoc
      */
-    public function setMetadata(?array $metadata): IngestionInterface
+    public function setMetadata(Metadata $metadata): IngestionInterface
     {
-        $this->metadata = $metadata;
+        $this->metadata[$metadata->getKey()] = $metadata->getValue();
         return $this;
     }
 
     /**
      * @inheritDoc
      */
-    public function getMetadata(): ?array
+    public function getMetadata(): Metadata
```

Comment:
$this->metadata is Metadata[]

```suggestion
    public function getMetadata(): array
    {
      return $this->metadata;
    }
```